src/logger/luaLoggerBinding.c

```diff
@@ -782,29 +784,83 @@ int Lua_FlashLoggerConfig(lua_State *L)
 
 int Lua_AddVirtualChannel(lua_State *L)
 {
-    size_t args = lua_gettop(L);
-    if (args < 2) return 0;
-
-    ChannelConfig cc;
-    strlcpy(cc.label, lua_tostring(L, 1), DEFAULT_LABEL_LENGTH);
-    cc.sampleRate = encodeSampleRate((unsigned short) lua_tointeger(L, 2));
-    cc.precision = args >= 3 ? lua_tointeger(L, 3) :
-                   DEFAULT_CHANNEL_LOGGING_PRECISION;
-    cc.min = args >= 4 ? lua_tointeger(L, 4) : DEFAULT_CHANNEL_MIN;
-    cc.max = args >= 5 ? lua_tointeger(L, 5) : DEFAULT_CHANNEL_MAX;
-    strlcpy(cc.units, args >=6 ? lua_tostring(L, 6) : DEFAULT_CHANNEL_UNITS,
-            DEFAULT_UNITS_LENGTH);
-    lua_pushinteger(L, create_virtual_channel(cc));
+        const size_t args = lua_gettop(L);
+        if (args < 2 || args > 6)
+                return incorrect_arguments(L);
 
-    return 1;
+        const char *label = lua_tostring(L, 1);
+        switch (validate_channel_config_label(label)) {
+        case CHAN_CFG_STATUS_OK:
+                break;
```

Comment:
suggest setting the label text on this line so it's consistent with how you do it on line 824